pkg/explorer/esplora/unspents.go

```diff
@@ -59,120 +77,141 @@ func (e *esplora) getUnspents(addr string, blindingKeys [][]byte) ([]explorer.Ut
 		return nil, fmt.Errorf("error on retrieving utxos: %s", err)
 	}
 
-	unspents := make([]explorer.Utxo, len(witnessOuts))
-	chUnspents := make(chan explorer.Utxo)
-	chErr := make(chan error, 1)
+	utxos := make([]explorer.Utxo, 0, len(witnessOuts))
+	chRes := make(chan utxoResult)
+	wg := &sync.WaitGroup{}
+	wg.Add(len(witnessOuts))
+
+	go func() {
+		wg.Wait()
+		close(chRes)
+	}()
 
 	for i := range witnessOuts {
 		out := witnessOuts[i]
-		go e.getUtxoDetails(out, chUnspents, chErr)
-
-		select {
-		case err := <-chErr:
-			if err != nil {
-				close(chErr)
-				close(chUnspents)
-				return nil, fmt.Errorf("error on retrieving utxos: %s", err)
-			}
+		go e.getUtxoDetails(out, chRes, wg)
+	}
 
-		case unspent := <-chUnspents:
-			if out.IsConfidential() && len(blindingKeys) > 0 {
-				go unblindUtxo(unspent, blindingKeys, chUnspents, chErr)
-				select {
+	utxosToUnblind := make([]explorer.Utxo, 0)
+	for r := range chRes {
+		if r.err != nil {
+			return nil, fmt.Errorf("error on retrieving utxos: %s", r.err)
+		}
 
-				case err := <-chErr:
-					close(chErr)
-					close(chUnspents)
-					return nil, fmt.Errorf("error on unblinding utxos: %s", err)
+		if len(blindingKeys) > 0 && r.utxo.IsConfidential() {
+			utxosToUnblind = append(utxosToUnblind, r.utxo)
+		} else {
+			utxos = append(utxos, r.utxo)
+		}
+	}
+
+	if len(utxosToUnblind) > 0 {
```

Comment:
i think we are losing performance in here since we are waiting to get details for all utxo's and only then we do unblinding, we should be able to do unblinding as soon as there is 1 utxo with details

Reply:
Consider that previously the execution was basically sequential even if we already use go routine s and channels... I don't think here we are loosing performance at all compared to what we have done before.

If you look at this form another perspective, If there's an error while getting the utxo details, we don't want to waste time in unblinding others because we return and abort all the process.

Reply:
not comparing it to previous version, just saying that we are not as 'parallel' as we could.

Reply:
I see.. I think I explained why we don't want to be as "concurrent" as we could though.